test/E2ETest/Microsoft.Test.E2E.AspNet.OData/ModelBuilder/RecursiveComplexTypesTests.cs

```diff
@@ -0,0 +1,489 @@
+                        }
+                    ]
+                }
+            ]
+        },
+        {
+            ""ID"": 1,
+            ""Customer"": {
+                ""Name"": ""Customer1"",
+                ""Accounts"": [
+                    {
+                        ""Number"": 0,
+                        ""Owner"": {
+                            ""Name"": ""Customer0"",
+                            ""Accounts"": []
+                        }
+                    }
+                ]
+            },
+            ""Address"": {
+                ""Street"": ""123 Main Street"",
+                ""City"": ""Seattle"",
+                ""Country"": ""USA"",
+                ""PreviousAddress"": {
+                    ""Street"": ""111 West 8th Avenue"",
+                    ""City"": ""Vancouver"",
+                    ""Country"": ""Canada"",
+                    ""PreviousAddress"": null
+                }
+            },
+            ""HomeDirectory"": {
+                ""Name"": ""/users/customer1"",
+                ""Size"": 2048,
+                ""Files"": [
+                    {
+                        ""Name"": ""users/customer1/file0"",
+                        ""Size"": 0
+                    },
+                    {
+                        ""Name"": ""users/customer1/subdir0"",
+                        ""Size"": 0,
+                        ""Files"": []
+                    },
+                    {
+                        ""Name"": ""users/customer1/subdir1"",
+                        ""Size"": 1024,
+                        ""Files"": [
+                            {
+                                ""Name"": ""users/customers1/subdir1/file0"",
+                                ""Size"": 0
+                            }
+                        ]
+                    }
+                ]
+            },
+            ""CustomFields"": [
+                {
+                    ""Name"": ""LastLoginDate"",
+                    ""DataType"": ""DateTime"",
+                    ""SubFields"": []
+                },
+                {
+                    ""Name"": ""Referrals"",
+                    ""DataType"": ""ComplexType"",
+                    ""SubFields"": [
+                        {
+                            ""Name"": ""Name"",
+                            ""DataType"": ""String"",
+                            ""SubFields"": []
+                        },
+                        {
+                            ""Name"": ""ReferralBonus"",
+                            ""DataType"": ""Double"",
+                            ""SubFields"": []
+                        }
+                    ]
+                }
+            ]
+        }
+    ]
+}";
+
+            string actualContent = await response.Content.ReadAsStringAsync();
+
+            AssertJsonEqual(ExpectedContent, actualContent);
+        }
+
+        private static void AssertJsonEqual(string expectedJson, string actualJson) =>
```

Comment:
> => [](start = 83, length = 3)

Can it pass the build?

Reply:
Apparently it did, although if you prefer I avoid using C# 6 features, I can do that.

Reply:
I'd prefer to using the new C# features. However, I can't make sure our build system can understand the new C# feature. So, Make sure the PR build pass. 

---
In reply to: [193254114](https://github.com/OData/webapi/pull/1458#discussion_r193254114) [](ancestors = 193254114)

Reply:
If you guys have any issues getting the build to pass with new features, you should be able to add Microsoft.Net.Compilers NuGet package to your project, and it should pick them up.

Reply:
Thanks.

---
In reply to: [193488586](https://github.com/OData/webapi/pull/1458#discussion_r193488586) [](ancestors = 193488586)